doc/7/getting-started/.react-native/screens/Chat/ChatClient.js

```diff
@@ -0,0 +1,149 @@
+/* snippet:start:1 */
+import React from 'react';
+import ChatView from './ChatView';
+import kuzzle from '../../services/kuzzle';
+/* snippet:end */
+
+/* snippet:start:2 */
+class ChatClient extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            messages: []
+        };
+        this.kuzzle = kuzzle;
+        this.handleSendMessage = this.onSendMessage.bind(this);
+    }
+
+    async componentDidMount() {
+        await this.initConnection();
+        await this.fetchMessages();
+        await this.subscribeMessages();
+    }
+
+    async initConnection() {
+        // Etablish the connection
+        await kuzzle.connect();
+        // Check if 'chat' index exists
+        if (! await kuzzle.index.exists("chat")) {
+            // If not, create 'chat' index and 'messages' collection
+            await kuzzle.index.create("chat");
+            await kuzzle.collection.create("chat", "messages");
+        }
+    }
+    /* snippet:end */
+
+    /* snippet:start:4 */
+    displayDate(previousDate, currentDate) {
+        if (previousDate === null) {// Message is the first of the array so need to display the date
+            return true;
+        }
+        const d1 = new Date(previousDate).toDateString();
+        const d2 = new Date(currentDate).toDateString();
+        if (d1 !== d2) {
+            // Previous message and current has different dates so need to display the date
+            return true;
+        };
+        // Previous message and current has same dates so doesn't need to display the date
+        return false;
+    }
+    /* snippet:end */
+
+    /* snippet:start:5 */
+    getMessage(msg, displayDate) {
+        const message = {
+            // The unique id of the document containing the message
+            id: msg._id,
+            // The text of the message
+            message: msg._source.message,
+            // The creation date
+            date: msg._source._kuzzle_info.createdAt,
+            // The author name
+            author: msg._source.author,
+            // Boolean to display or not the date
+            displayDate: displayDate
```

Comment:
Prefer usage of shorthand property
```suggestion
            displayDate
```